cmd/cosign/cli/verify_manifest.go

```diff
@@ -116,95 +110,91 @@ func (c *VerifyManifestCommand) Exec(ctx context.Context, args []string) error {
 	return c.VerifyCommand.Exec(ctx, images)
 }
 
-func getImagesFromYamlManifest(manifest []byte) ([]string, error) {
-	dec := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(manifest), 4096)
-	cScheme := runtime.NewScheme()
-	var images []string
-	if err := corev1.AddToScheme(cScheme); err != nil {
-		return images, err
-	}
-	if err := appsv1.AddToScheme(cScheme); err != nil {
-		return images, err
-	}
-	if err := batchv1.AddToScheme(cScheme); err != nil {
-		return images, err
-	}
-
-	deserializer := serializer.NewCodecFactory(cScheme).UniversalDeserializer()
-	for {
-		ext := runtime.RawExtension{}
-		if err := dec.Decode(&ext); err != nil {
-			if errors.Is(err, io.EOF) {
-				break
+// imageContainer is the union type that match PodSpec, PodSpecTemplate, and
+// JobSpecTemplate; but filtering all keys except for `Image`.
+type imageContainer struct {
+	Spec struct {
```

Comment:
I'd be curious what this did if you just inlined both of `duckv1.Pod` and `duckv1.WithPod` 🤔  😅  🤣

Reply:
I did try, and you can't because both `duckv1.Pod` and `duckv1.WithPod` define the common metadata and only one wins. The decoder gets real confused 🤣 .

Reply:
I could create my own union type and break the `duckv1.Pod` components up and make a new parent wrapper like I do here, but this usage just is not complicated enough to bother

Reply:
Yeah, I wouldn't worry about it.